### ISSUE TITLE
ActionCable Redis adapter ssl_params like `cert` and `key` are not recognised

### ISSUE THREAD

**Issue by tsrivishnu** [rails/rails#50209]
### Steps to reproduce
* Configure ActionCable Redis adapter with `ssl: true` and `ssl_params` that contain paths to `cert` and `key` files.
   ```yaml
    # config/cable.yml
    development:
      adapter: redis
      url: "redis://localhost:6379/0"
      ssl: true
      ssl_params:
        ca_file: config/credentials/redis/dev/ca.crt
        cert: config/credentials/redis/dev/client.crt
        key: config/credentials/redis/dev/client.key
      channel_prefix: app_cable_development

   ```
* Open rails console and try to broadcast a message
   ```ruby
    # In rails console
    ActionCable.server.broadcast("hi", "hello world")
* Fails with error exception
    ```
    => 2023-11-29 11:48:08.317258 D [7:6040 broadcasting.rb:43] ActionCable [ActionCable] Broadcasting to hi: "hello world"
    /app/.bundle/ruby/3.1.0/gems/redis-client-0.12.1/lib/redis_client/ruby_connection.rb:59:in `initialize': wrong argument type String (expected OpenSSL/X509) (TypeError)
    ```

### Expected behavior
The message is broadcasts without any error,

### Actual behavior
Fails with error:
```
/app/.bundle/ruby/3.1.0/gems/redis-client-0.12.1/lib/redis_client/ruby_connection.rb:59:in `initialize': wrong argument type String (expected OpenSSL/X509) (TypeError)
```

### System configuration
**Rails version**: 7.0.4

**Ruby version**: 3.10

**Comment by tsrivishnu** [comment]
After some debugging, I realised the issue to be the following: 

ActionCable::SubscriptionAdapter::Redis#redis_connection calls `symbolize_keys` on the config. That symbolises the root level keys but not the keys inside `ssl_params`. 

For `redis-client` to read files from path values, the keys need to be symbols: https://github.com/redis-rb/redis-client/blob/v0.18.0/lib/redis_client/ruby_connection.rb#L17

As a work around, I added an initialiser overriding `.redis_connector`.

```ruby
# config/initializers/action_cable.rb
require "action_cable/subscription_adapter/redis"

ActionCable::SubscriptionAdapter::Redis.redis_connector = lambda do |config|
  ::Redis.new(config.deep_symbolize_keys.except(:adapter, :channel_prefix))
end
```

**Comment by tsrivishnu** [comment]
This seem to be fixed in the latest versions: f1ace639736ae49ccf9468aec3f6a478b77d1744. Will close now.
